hypertrace-graphql-gateway-service-metric-utils/src/main/java/org/hypertrace/graphql/utils/metrics/gateway/MetricAggregationMapConverter.java

```diff
@@ -1,45 +1,95 @@
 import org.hypertrace.gateway.service.v1.common.AggregatedMetricValue;
 import org.hypertrace.graphql.metric.request.MetricAggregationRequest;
-import org.hypertrace.graphql.metric.schema.MetricAggregation;
+import org.hypertrace.graphql.metric.schema.BaselinedMetricAggregation;
+import org.hypertrace.graphql.metric.schema.Health;
+import org.hypertrace.graphql.metric.schema.MetricBaselineAggregation;
 
 class MetricAggregationMapConverter
-    implements BiConverter<
+    implements TriConverter<
         List<MetricAggregationRequest>,
         Map<String, AggregatedMetricValue>,
-        Map<MetricLookupMapKey, MetricAggregation>> {
+        Map<String, Baseline>,
+        Map<MetricLookupMapKey, BaselinedMetricAggregation>> {
 
   private final MetricAggregationConverter aggregationConverter;
+  private final MetricBaselineAggregationConverter baselineAggregationConverter;
 
   @Inject
-  MetricAggregationMapConverter(MetricAggregationConverter aggregationConverter) {
+  MetricAggregationMapConverter(
+      MetricAggregationConverter aggregationConverter,
+      MetricBaselineAggregationConverter baselineAggregationConverter) {
     this.aggregationConverter = aggregationConverter;
+    this.baselineAggregationConverter = baselineAggregationConverter;
   }
 
   @Override
-  public Single<Map<MetricLookupMapKey, MetricAggregation>> convert(
+  public Single<Map<MetricLookupMapKey, BaselinedMetricAggregation>> convert(
       List<MetricAggregationRequest> aggregationRequests,
-      Map<String, AggregatedMetricValue> resultMap) {
+      Map<String, AggregatedMetricValue> resultMap,
+      Map<String, Baseline> baselineMap) {
     return Observable.fromIterable(aggregationRequests)
         .flatMapSingle(
-            aggregationRequest -> this.buildMetricAggregationEntry(aggregationRequest, resultMap))
+            aggregationRequest ->
+                this.buildMetricAggregationEntry(aggregationRequest, resultMap, baselineMap))
         .collect(immutableMapEntryCollector());
   }
 
-  private Single<Entry<MetricLookupMapKey, MetricAggregation>> buildMetricAggregationEntry(
-      MetricAggregationRequest aggregationRequest, Map<String, AggregatedMetricValue> resultMap) {
-    return this.aggregationConverter
-        .convert(resultMap.get(aggregationRequest.alias()).getValue())
+  private Single<Entry<MetricLookupMapKey, BaselinedMetricAggregation>> buildMetricAggregationEntry(
+      MetricAggregationRequest aggregationRequest,
+      Map<String, AggregatedMetricValue> resultMap,
+      Map<String, Baseline> baselineMap) {
+    return zip(
+            this.aggregationConverter.convert(resultMap.get(aggregationRequest.alias()).getValue()),
+            this.baselineAggregationConverter.convert(
+                Optional.ofNullable(baselineMap.get(aggregationRequest.alias()))),
+            (metricAggregation, baselinedMetricAggregation) ->
+                new BaselinedMetricAggregation() {
```

Comment:
Can we make real (inner or otherwise) classes and remove the inline anonymous class? Makes the code much harder to follow